crates/iota-data-ingestion-core/src/history/reader.rs

```diff
@@ -0,0 +1,313 @@
+    ///
+    /// Returns an error in the following cases:
+    ///
+    /// * If fetching the file from the remote store fails.
+    /// * If the file is corrupted and fails to decode.
+    pub async fn iter_for_file(
+        &self,
+        file_path: Path,
+    ) -> Result<impl Iterator<Item = CheckpointData>> {
+        let raw_data_batch = get(&self.remote_object_store, &file_path).await?;
+        let data_batch = make_iterator::<CheckpointData, Reader<Bytes>>(
+            CHECKPOINT_FILE_MAGIC,
+            raw_data_batch.reader(),
+        )?;
+        Ok(data_batch)
+    }
+
+    /// Return latest available checkpoint in archive.
+    pub async fn latest_available_checkpoint(&self) -> Result<CheckpointSequenceNumber> {
+        self.manifest
+            .lock()
+            .await
+            .next_checkpoint_seq_num()
+            .checked_sub(1)
+            .ok_or_else(|| {
+                IngestionError::HistoryRead("no checkpoint data in the remote store".into())
+            })
+    }
+
+    pub fn remote_store_identifier(&self) -> String {
+        self.remote_object_store.to_string()
+    }
+
+    /// Syncs the Manifest from remote store.
+    pub async fn sync_manifest_once(&self) -> Result<()> {
+        Self::sync_manifest(self.remote_object_store.clone(), self.manifest.clone()).await?;
+        Ok(())
+    }
+
+    pub async fn get_manifest(&self) -> Manifest {
+        self.manifest.lock().await.clone()
+    }
+
+    /// Copies Manifest from remote store to the given Manifest.
+    async fn sync_manifest(
+        remote_store: Arc<dyn ObjectStoreGetExt>,
+        manifest: Arc<Mutex<Manifest>>,
+    ) -> Result<()> {
+        let new_manifest = read_manifest(remote_store.clone()).await?;
+        let mut locked = manifest.lock().await;
+        *locked = new_manifest;
+        Ok(())
+    }
+
+    /// Resolve the files to fetch for the specified range.
+    ///
+    /// The method retrieves the manifest from the remote store and
+    /// searches for the files that cover the given range of checkpoint
+    /// data.
+    ///
+    /// # Errors
+    ///
+    /// The method fails if the remote store has no data, or if the
+    /// manifest fails to verify.
+    async fn get_files_for_range(
+        &self,
+        checkpoint_range: Range<CheckpointSequenceNumber>,
+    ) -> Result<impl Iterator<Item = FileMetadata>> {
+        let manifest = self.get_manifest().await;
+
+        let latest_available_checkpoint = manifest
+            .next_checkpoint_seq_num()
+            .checked_sub(1)
+            .ok_or_else(|| {
+                IngestionError::HistoryRead("no checkpoint data in the remote store".into())
+            })?;
+
+        if checkpoint_range.start > latest_available_checkpoint {
+            return Err(IngestionError::HistoryRead(format!(
+                "latest available checkpoint is: {latest_available_checkpoint}",
+            )));
+        }
+
+        let files = self.verify_manifest(manifest)?;
```

Comment:
The point of calling `self.verify_manifest` seems to be getting the files from it. Isn't the name `verify_manifes` a bit misleading?